### ISSUE TITLE
POST Read Callback never called.

### ISSUE THREAD

**Issue by bagder** [curl/curl#7359]
### Discussed in https://github.com/curl/curl/discussions/7356

<div type='discussions-op-text'>

<sup>Originally posted by **jshanab** July  7, 2021</sup>
I have been using curl to implement rtsp-over-http/s for years. We have been using 7.48.x and the Post was using CURLOPT_READFUNCTION and CURLOPT_READDATA eith a content length set to a very very large value.

The protocol is such that you open the post channel and it stays open and all rtsp commands are sent base 64 encoded. 
We use the Pause and Resume to handle this and once video is streaming on the GET channel, a command is only sent every 5 seconds until the TEARDOWN is sent.

We are updating to 7.77 and while the posts with fields or the posts with known lengths still work. This Post with a large size, to allow ongoing posts, has stopped working. In fact I cannot get it to call the readback function at all. 

I have tried changing to Chunked transfer because I went thru the change log and that seems like that might of changed in 7.66,   but I cannot get it to call the read callback.  

What are the conditions to call and more importantly what conditions prevent curl from calling the ReadFunction.

Thanks
</div>

**Comment by bagder** [comment]
Thanks a lot for your report!

I've failed to reproduce this problem myself, so can I ask you to provide us with more details on how to go ahead and repeat this. Preferably with a command line or a stand-alone program we can run from our ends against a public URL to trigger the problem?

**Comment by jshanab** [comment]
This issue can be closed as " not an issue" (CURLM_RECURSIVE_API_CALL ERROR)
It was on my side. Moral of story, ALWAYS check the curl return code.
       
I created a minimal test and it, of course, worked great.
Which lead me to dig deeper into our framework where we adding the easy handle to the multihandle and found we were not checking the return code there. When I did, I found I was triggering the CURLM_RECURSIVE_API_CALL ERROR.

It made sense to do this protocol this way and 7.48 lets us get away with it for years. We were just creating the next request in the callback.  

Thanks.
